server/src/main/java/org/elasticsearch/common/blobstore/BlobContainer.java

```diff
@@ -37,39 +37,55 @@ public interface BlobContainer {
     /**
      * Tests whether a blob with the given blob name exists in the container.
      *
-     * @param   blobName
-     *          The name of the blob whose existence is to be determined.
-     * @return  {@code true} if a blob exists in the {@link BlobContainer} with the given name, and {@code false} otherwise.
+     * @param purpose The purpose of the operation, useful for stats collection.
+     * @param blobName The name of the blob whose existence is to be determined.
+     * @return {@code true} if a blob exists in the {@link BlobContainer} with the given name, and {@code false} otherwise.
      */
-    boolean blobExists(String blobName) throws IOException;
+    boolean blobExists(OperationPurpose purpose, String blobName) throws IOException;
+
+    @Deprecated(forRemoval = true)
+    default boolean blobExists(String blobName) throws IOException {
+        return blobExists(OperationPurpose.SNAPSHOT, blobName);
+    }
 
     /**
      * Creates a new {@link InputStream} for the given blob name.
      *
-     * @param   blobName
-     *          The name of the blob to get an {@link InputStream} for.
-     * @return  The {@code InputStream} to read the blob.
-     * @throws  NoSuchFileException if the blob does not exist
-     * @throws  IOException if the blob can not be read.
+     * @param purpose The purpose of the operation, useful for stats collection.
+     * @param blobName The name of the blob to get an {@link InputStream} for.
+     * @return The {@code InputStream} to read the blob.
+     * @throws NoSuchFileException if the blob does not exist
+     * @throws IOException         if the blob can not be read.
      */
-    InputStream readBlob(String blobName) throws IOException;
+    InputStream readBlob(OperationPurpose purpose, String blobName) throws IOException;
+
+    @Deprecated(forRemoval = true)
+    default InputStream readBlob(String blobName) throws IOException {
+        return readBlob(OperationPurpose.SNAPSHOT, blobName);
+    }
```

Comment:
Similarly, methods marked as deprecated in this class will be removed once the necessary changes are made in the other repo. They are here to avoid breaking builds for now.